encoding/m3tsz/encoder.go

```diff
@@ -315,14 +507,13 @@ func (enc *encoder) Stream() xio.SegmentReader {
 		tail = scheme.Tail(b[blen-1], pos)
 	}
 
-	segment := ts.Segment{Head: head, Tail: tail, TailShared: true}
 	readerPool := enc.opts.GetSegmentReaderPool()
 	if readerPool != nil {
 		reader := readerPool.Get()
-		reader.Reset(segment)
+		reader.Reset(ts.Segment{Head: head, Tail: tail})
```

Comment:
Why removing `TailedShared: true`?  This will cause the tail to be returned a bytes pool.